### DISCUSSION TITLE
+++ New REV2 HW released+++

### DISCUSSION THREAD

**Discussion by xyphro** [xyphro/UsbGpib#46] [Announcements]
It is done: after succesful building and testing 50 prototypes I feel confident that the new design works as good as the previous one while bringing improvements on mechanis and boot loader entering.

I had also convinced myself that a USB type C connector is the better choice, ignoring the inner devil in me which thinks that bigger connectors are mechanically more stable :-)

Also a new firmware image is released for beta testing. Please feedback your experience with it to me - only when I get to know about issues I can fix them and only when I know others have good results I get confidence to baseline.

Have fun building and using it!

**Comment by makeitsquare** [comment]
Awesome! Thanks a lot. I just ordered some PCBs form JLCPCB (five PCBs with two assembled boards). Let's hope that I picked a fitting USB-C connector. I am happy to share the BOM and CPL (placement) file for JLCPCB if someone is interested. I'll report back when I receive the boards.

**Reply by xyphro** [reply]
Ah, forgot to add mine. Have added my BOM and CPL files...

**Reply by makeitsquare** [reply]
Hi,

I just received my boards from JLCPCB. It took my quite a while to get them running. At first, my Atmel ISP programmer (one of those cheap clones) seem to have failed and I had to get a STM32 blue pill board working with FASTUSBasp so that I could flash Bootloader.hex. This would have been a bit easier if the pin-out of the ISP header would be included in the Readme somewhere. I understand that the pin-out is the standard Atmel ISP pinout, but it took me some time to hook it up correctly.

Pin-out is:
Pin Signal
1    MISO
2   +5V   
3   SCK
4   MOSI
5   RST
6   GND

Pins 1, 3 and 5 are on the top of the board and 2,4,6 on the bottom.

After flashing the boot loader, the boards showed up as flash drives under Windows 10, Linux (Ubuntu 20.04) and MacOS 14.2.1. But copying the TestAndMeasurement.bin only seems to work correctly with Windows? I tried both cp and dd as shown in the Readme under Linux and MacOS, but the board seemed to be locked / dead afterwards. Only reflashing the boot loader made them work again (pressing the button and connecting to a working GPIB device didn't trigger any USB enumeration on MacOS or Linux...). Once I copied the bin with Windows the boards started to work. Not sure what is going on there...

Finally, I printed your housing on my 3d printer (took 1h30m on my modified Anet E10). Unfortunately, the opening for the button seemed to be a bit too small (perhaps I picked the wrong button from JLCPCB), so one button came of when I tried to press the PCB into the housing. Nothing a bit of knife rework couldn't fix, but perhaps someone reading this is a bit more careful than me ;)

Thanks again, I can confirm that REV2 with the ReleaseCandidate FW is working with MacOS and my HP 3478A!

**Reply by xyphro** [reply]
Yeah, that's the beauty of the 50 different programmer variants for AVR cores and 20 different choices of programmer software for it :-)
You might want to decrease the ISP clock frequency for programming for some of those as they use a too high one default wise and depending which one you have. AVRs have the limitation of coreclock/8 as max frequency and 1MHz is too high and just on the edge of working depending on how the programmer programs it (fuse deletion, flash erasing, ... order specifically). Some also come with fancy non standard pinouts requiring jumperwirings instead of 1:1 cables. This breaks the concept of using standard pinouts.

On the print: Also here it is the beauty of 3d printing - each printer has different tolerances, some make blobs in corners where they slow down, some don't, some compress circles slightly to Ovals (bad corexy ones) some slicers settings make the walls uneven, ...
I used 2 of my own printers so far and had ordered 3d prints from JLCPCB and they fit. Which does not mean that this is for anyone the case as there are many variables. From a commercial 3d print services from France I got prints that are so crappy that I threw them in the bin after receiving them.

I never broke a switch so far, but can increase the cavity at this spot a bit to reduce the risk.

Anyway: Glad you made it!

Ps: You can improve documentation with a pull request if you like to see specific points improved. Especially the programming is something where I have a hard time to structure it properly considering that there are many programmer options to choose from and I have only 3 variants.

**Reply by xyphro** [reply]
Just checked the stl... can you maybe show me which part made it break and share the jlc partnumber of the switch you used?

**Reply by makeitsquare** [reply]
See the pictures attached. The switch was too high and hitting the housing. So I exacto'd the housing until it fit ;) The print wasn't the cleanest, but I had to cut at least 1 mm or a bit more. The switch is approx. 3.5 mm high (without the button on top). JLCPCB part number is C118141 (https://www.lcsc.com/product-detail/Tactile-Switches_BZCN-TSB013A5026B_C2888463.html) 
![IMG_4368](https://github.com/xyphro/UsbGpib/assets/84725204/3422ce27-8952-44d4-bfd8-fd18faa43a89)
![IMG_4367](https://github.com/xyphro/UsbGpib/assets/84725204/194defd6-34ff-4e13-b0ca-f11ad33e0ce8)

**Reply by xyphro** [reply]
Ahh, I see - the switch you mounted is much higher as the one I used. It is hard to see, but it feels, that the push button part is raising above the height of the USB connector.

My switch measures ~2.5mm in total height (top of button down towards PCB top side).

Here for reference a side view of one of my adapters:
![grafik](https://github.com/xyphro/UsbGpib/assets/4944408/10d6143e-75bc-4f44-8dc7-26add2aee6cf)

I had uploaded my CPL file in the HW/REV2 folder in case you want to check the JLC Part# I used.

Good that we engineers have McGuyver skills to work such things :-D

Thanks for posting the pictures!

**Comment by Niravk1997** [comment]
Have you tested the new R&S Visa 7.2.2 software for windows? That version couldn't recognize the adapter in the Resource Manager. The adapter would show up in the device manager with drivers installed but won't be visible to any software. I had to revert back to R&S Visa 5.12.3. Maybe I need to reformat windows and reinstall the visa software.

**Reply by xyphro** [reply]
I have this with any usbtmc device and R&S visa. On one PC, but not on another.

**Reply by xyphro** [reply]
![grafik](https://github.com/xyphro/UsbGpib/assets/4944408/7b381174-e459-4f66-83cb-d9e4999358c7)

this is what I get in the log view when trying to search fo devices. As said: This is independent of the USBGPIB adapter and also e.g. my RTM2032 is not found using that version of R&S Visa. Also VXI-11 devices are not found. The R&S Visa same version works well on another PC. 
Will reach later out to R&S after some more analysis.

**Reply by xyphro** [reply]
BTW: I have a mixed NI VISA and R&S Visa installation. When choosing NI Visa as visa implementation within tester64 application, device search is working:
![grafik](https://github.com/xyphro/UsbGpib/assets/4944408/a2a8b797-9358-46e0-804b-b8852e3fb011)

I never had issues finding devices using python using R&S Visa, so I suspect it is a Tester64 application related issue.

**Comment by xyphro** [comment]
Hi Nirav,

R&S has very quickly addressed this and released R&S Visa 7.2.3. Searching Instruments works now fine - just tested it!

If I would not need the NI GPIB adapter sometimes fpr crosschecks, I would now deinstall NI Visa :-)

Best regards,

Kai

**Reply by Niravk1997** [reply]
Yes, I have just tested the new version. It does recognize the devices. Thank you for contacting them and having the issue resolved.

![image](https://github.com/xyphro/UsbGpib/assets/52579388/6f7a8a23-b28a-4a58-ac2d-3815016db0f4)

Sincerely,
Nirav

**Comment by Niravk1997** [comment]
Hey xyphro,

Something I noticed with the new firmware is that once the instrument is shutdown (powered off completely) the led on the adapter still remains solid (not blinking) and the instrument still remains in the device manager. With the older firmware, when the instrument is powered off, the adapter’s led will start blinking and the instrument will disappear from the device manager. Can you look into this and see if you can reproduce this issue.

![image](https://github.com/xyphro/UsbGpib/assets/52579388/35a8640a-d89a-4fc3-9fe3-1b0d5d1f4b40)

Sincerely,
Nirav

**Reply by xyphro** [reply]
Oh, that is indeed not intended behavior. Will check and fix!

Update: yes, I have the same behavior. How can I have missed it? Will fix it.

**Reply by xyphro** [reply]
Hi Nirav,

can you check if the issue is solved for you too with attached firmware?

If not: Is the 100kOhm pulldown on the ATN present? I know stupid question as you say it is Firmware version related, still just want to be sure...

It is also time for a new update in general with the changes that Daz100 suggested on the AutoId/termination and other configuration items and I recently implemented. So once you confirm it is solved, I will push a general update again...

[TestAndMeasurement_FixNiravsIssue.zip](https://github.com/xyphro/UsbGpib/files/13907923/TestAndMeasurement_FixNiravsIssue.zip)

Best regards,

Kai

**Reply by Niravk1997** [reply]
Hey xyphro,

So I just tested this firmware and the issue has not been solved. After completely powering the instrument down the adapter still shows up in the device manage and Visa find device query. You can even connect to device but since the device is powered down a visa timeout occurs.

Also, yes the 100kOhm pulldown on the ATN is present on the adapter. I tested both versions, the one your uploaded here and the one in the github that you just updated.

Is this issue not present on your end? Try powering down your instrument after the adapter has initialized properly and see if the led starts blinking or remains solid.

Sincerely,
Nirav

**Reply by xyphro** [reply]
It was present in a random way. Sometimes it did not disconnect, sometimes it did.
With the fix it was solved for me.

I changed the presence detection slightly again. Can you try again? (FW VERSION readout is incremented to V1.7).

Tried it successfully on: CMU200, HP5335a and DM199. In case it still fails I go and get my bulky TDS7104 cause I guess you use a Tek instrument.

[TestAndMeasurement_Try2.zip](https://github.com/xyphro/UsbGpib/files/13932791/TestAndMeasurement_Try2.zip)

Note that the cause is the introduction of the introduction of the automatic serial polling on SRQ sensing. That complicated the presence detection a bit. Hope it is ok now.

**Reply by Niravk1997** [reply]
Yes, second time's the charm. The issue is gone. Tested on HP34401a, HP3478A and Keithley 2304a for now. For all instruments, after powering down, the adapter started blinking and the device manager no longer shows the devices connected.

Sincerely,
Nirav

**Reply by xyphro** [reply]
Perfect, the morning starts good. Thanks for testing Nirav!

**Reply by xyphro** [reply]
I have committed the update to Master right now. So from now onwards the fix you tested yesterday is baselined.

**Comment by rjordans** [comment]
Thanks for posting this project.  I have built 5 rev-2 boards so far and they all work fine.

Tested with Linux using
- Rohde & Schwarz SMY01
- Fluke 6080A
- HP/Agilent 33120A
- Boonton 4200
- HP 853A
- HP 4274A

Tested with a Macbook using
- Rohde & Schwarz SML03

No problems so far, although the older equipment doesn't like the autoid function so that's now disabled on my adapters.  The tested adapters also seem to work fine over a few meters of GPIB cable.

**Reply by xyphro** [reply]
Geweldig Roel, dank je wel voor het deelen. Dit is leuk voor anderen om te zien of iets compatibel is of niet!

**Comment by wa1hco** [comment]
I have a question about the programming pads in the corner.   I saw the photo of a 6 pin dual row header soldered at 45 deg...ok, I guess, but needs to be removed after programming.  What was Xyphro's intent with that "pins in the corner" idea?  Is there a push on connector?

**Reply by xyphro** [reply]
The PCB dimensions did not allow for a normal connector, so I decided to come up with that solution. As mentioned in the hw/rev2 document the intend is to solder once 2 3-pin pinheaders there to get a standard 6 pin ISP connector. After programming it can be quickly removed.

All other solutions ended up in increasing the PCB size which I didn't want for something which is just required once.

I personally have build a spring contact based adapter to program batches of PCBs quickly.

Another user had a good suggestion which is using a 6 pin (2x3) 2mm pitch pinheader and push it on the PCB to make contact (tried it too, it works really surprisingly well)

**Reply by xyphro** [reply]
I just came across a photo of using a 2mm pinheader. In this case the 2mm pinheader is too big, but shows the principle.
You can also use a 2mm pinheader as is without adapter PCB.
I think this is an elegant solution (credits to Jörg Wünsch from Microcontroller.net).

![image](https://github.com/user-attachments/assets/da0cda6f-bfea-4bbd-9223-3adb016189e8)

**Reply by xyphro** [reply]
And this was my solution. Not so simple, but made for the purpose of programming multiple boards one after each other quickly. A bit ugly, but it was a 30 minutes quick hack made out of bare PCB, cyanoacrylate and a cutting knife. And such quick hack temporary solution last... forever, right? :-)
For just 1 pieces such a solution doesn't make sense effort wise. The spring contacts I used here are extracted from a female USB-Type B connector.

![image](https://github.com/user-attachments/assets/7eeb935c-2a02-4988-a04c-a2e518b1d344)
![image](https://github.com/user-attachments/assets/678019fd-14c6-4d98-802e-0fed0212cb77)

**Comment by wa1hco** [comment]
Several good ideas, here.

Thanks,

Jeff
